hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/utils/TestRowDataToAvroConverters.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utils;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.flink.formats.common.TimestampFormat;
+import org.apache.flink.formats.json.JsonToRowDataConverters;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.hudi.util.AvroSchemaConverter;
+import org.apache.hudi.util.RowDataToAvroConverters;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.TimeZone;
+
+import static org.apache.flink.table.api.DataTypes.ROW;
+import static org.apache.flink.table.api.DataTypes.FIELD;
+import static org.apache.flink.table.api.DataTypes.TIMESTAMP;
+
+class TestRowDataToAvroConverters {
+
+  DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+  @Test
+  void testRowDataToAvroStringToRowDataWithLocalTimezone1() throws JsonProcessingException {
+    TimeZone.setDefault(TimeZone.getTimeZone(ZoneId.of("Asia/Shanghai")));
+    String timestampFromUtc8 = "2021-03-30 15:44:29";
+
+    DataType rowDataType = ROW(FIELD("timestamp_from_utc_8", TIMESTAMP()));
+    JsonToRowDataConverters.JsonToRowDataConverter jsonToRowDataConverter =
+            new JsonToRowDataConverters(true, true, TimestampFormat.SQL)
+            .createConverter(rowDataType.getLogicalType());
+    Object rowData = jsonToRowDataConverter.convert(new ObjectMapper().readTree("{\"timestamp_from_utc_8\":\"" + timestampFromUtc8 + "\"}"));
+
```

Comment:
OK，I will add ITs

Reply:
And why we just use the `TIMESTAMP_LTZ` instead of `TIMESTAMP` when local time zone is needed?

Reply:
I think this pr is a bug fix. In other RowDataToXXX classes in flink, local timezone is used to convert TIMESTAMP field instead of utc timezone, e.g. [RowDataToCsv](https://github.com/apache/flink/blob/release-1.18.0/flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/RowDataToCsvConverters.java#L308), [RowDataToJson](https://github.com/apache/flink/blob/a5548ccc21d85288c111f6be9df36689a11f9f0d/flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/RowDataToJsonConverters.java#L187), [ParquetWriter](https://github.com/apache/flink/blob/a5548ccc21d85288c111f6be9df36689a11f9f0d/flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/row/ParquetRowDataWriter.java#L483)

Reply:
This issue can also be found in [RowDataToAvro](https://github.com/apache/flink/blob/5fe6f2024f77c03b2f2a897827765b209c42d5b0/flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/RowDataToAvroConverters.java#L194) class in flink. And I find a [related pr](https://github.com/apache/flink/pull/23511) to fix it.

Reply:
IT is added in ITTestHoodieDataSource.